ie/source/Logger.cpp

```diff
@@ -112,30 +97,275 @@ void Logger::write(const std::wstring& message, Logger::Level level)
+                    #ifdef LOGGER_TIMESTAMP
+                        timestampOnly(fsBg);
+                    #endif // LOGGER_TIMESTAMP
+                    fsBg << message << std::endl << std::flush;
+                    fsBg.close();
+                } else if (level == Logger::FG) {
+                    std::wofstream fsFg;
+                    fsFg.open(m_fgfilename, std::ios::out | std::ios::app);
+                    #ifdef LOGGER_TIMESTAMP
+                        timestampOnly(fsFg);
+                    #endif // LOGGER_TIMESTAMP
+                    fsFg << message << std::endl << std::flush;
+                    fsFg.close();
+                } else if (level == Logger::SYS) {
+                    std::wofstream sysFg;
+                    sysFg.open(m_sysfilename, std::ios::out | std::ios::app);
+                    #ifdef LOGGER_TIMESTAMP
+                        timestampOnly(sysFg);
+                    #endif // LOGGER_TIMESTAMP
+                    sysFg << message << std::endl << std::flush;
+                    sysFg.close();
+                }
+            }
+        }
+    }
+}
 
-        if (m_filename != L"") {
-            std::wofstream fs;
-            fs.open(m_filename, std::ios::out | std::ios::app);
-#ifdef LOGGER_TIMESTAMP
-            timestamp(fs);
-#endif // LOGGER_TIMESTAMP
-            if (level == Logger::ERR) {
-                fs << L"[ERROR] ";
+void Logger::writeOnTab(const std::wstring& message, const std::wstring& onTabId)
+{
+    if (!this->enabled) {
+        return;
+    }
+
+    if (m_tablogfolder != L"") {
+        try {
+            std::wofstream fsTab;
+            fsTab.open(m_tablogfolder + L"\\extension-tab-" + onTabId + L".log", std::ios::out | std::ios::app);
+            #ifdef LOGGER_TIMESTAMP
+                timestampOnly(fsTab);
+            #endif // LOGGER_TIMESTAMP
+            fsTab << message << std::endl << std::flush;
+            fsTab.close();
+        }
+        catch (...) {
+            this->debug(L"EX: Could not write " + message + L" on tab" + onTabId);
+        }
+    }
+}
+
+void Logger::logIESetting(LPCTSTR hValueName) {
+    std::wstring errorLog;
+    try {
+        HKEY hKey;
+        wchar_t lszValue[512];
+        DWORD dwType = REG_SZ;
+        DWORD dwSize = 512;
+        if (::RegOpenKeyEx(HKEY_CURRENT_USER, TEXT("SOFTWARE\\Microsoft\\Internet Explorer\\Main"), 0, KEY_QUERY_VALUE, &hKey) == ERROR_SUCCESS)
+        {
+            std::wstring strLog = hValueName;
+            if (::RegQueryValueEx(hKey, hValueName, NULL, &dwType, (LPBYTE)&lszValue, &dwSize) == ERROR_SUCCESS)
+            {
+                strLog.append(L" = ");
+                strLog.append(lszValue);
+                this->logSystem(strLog);
             }
-            fs << message << std::endl << std::flush;
-            fs.close();
+            else
+            {
+                strLog.append(L" = ");
+                strLog.append(L"TBD");
+                this->logSystem(strLog);

```

Comment:
line 187 and 193 could be moved outside of the if, same with lines 185 and 191:

strLog.append(L" = ");
if (...)
{
  strLog.append(lszValue);
}
else 
{
  strLog.append(L"TBD");
}
this->logSystem(strLog);